### ISSUE TITLE
'notmuch_database_open' is deprecated

### ISSUE THREAD

**Issue by szoemi** [tsto/notmuchfs#16]
cc -g -O2 -std=c99 -Wall -Wextra -Werror -D_FILE_OFFSET_BITS=64   -c -MD -o notmuchfs.o notmuchfs.c
notmuchfs.c: In function 'database_open':
notmuchfs.c:215:6: error: 'notmuch_database_open' is deprecated: function deprecated as of libnotmuch 5.4 [-Werror=deprecated-declarations]
  215 |      notmuch_database_open(global_config.mail_dir,
      |      ^~~~~~~~~~~~~~~~~~~~~
In file included from notmuchfs.c:66:
/usr/include/notmuch.h:332:1: note: declared here
  332 | notmuch_database_open (const char *path,
      | ^~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make: *** [Makefile:37: notmuchfs.o] Error 1

**Comment by keithbowes** [comment]
Yeah, looking at notmuch.h, it seems that it was replaced by `notmuch_database_open_with_config` in libnotmuch 5.4 (notmuch 0.32). It should be easy to use the `LIBNOTMUCH_CHECK_VERSION` macro to see which to use.
